### PR TITLE
HRINT-2695  - IndexCompareExchange as java

### PR DESCRIPTION
**Pull request by Drorlev**
IndexCompareExchange as java

### CODE REVIEW

**Comment by Drorlev** [comment]
This is just the code

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/IndexCompareExchange.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                return unitsInStock;
+            }
+
+            public void setUnitsInStock(int unitsInStock) {
+                this.unitsInStock = unitsInStock;
+            }
+        }
+
+        public Products_ByUnitsInStock() {
+            /*
```
Comment (ml054) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/IndexCompareExchange.java:31]: ?

**Comment by Drorlev** [comment]
Ready for full review

**Review by ml054 — APPROVED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/metadata.json [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                {
+                    "Title": "Index Compare-Exchange values",
+                    "Type": "link",
+                    "Url": "https://ravendb.net/docs/article-page/latest/Csharp/indexes/indexing-compare-exchange-values#querying-the-index"
+                },
+                {
+                    "Title": "Querying an index",
+                    "Type": "link",
+                    "Url": "https://ravendb.net/docs/article-page/latest/csharp/indexes/querying/basics#example-iv---querying-a-specified-index"
+                }
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/metadata.json:156]: all above 3 links go to C#
pls fix

**Review by Danielle9897 — CHANGES_REQUESTED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/IndexCompareExchange.java (hunk 1 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/IndexCompareExchange.java]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/IndexCompareExchange.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        List<Product> products;
+
+        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
+            //region Step_4
+            products = session.query(Products_ByUnitsInStock.IndexEntry.class, Products_ByUnitsInStock.class)
+                .whereGreaterThan("UnitsInStock" , minValue)
+                .ofType(Product.class)
+                .toList();
+            //endregion
+            //endregion
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/IndexCompareExchange.java:56]: align with `//region demo`

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/metadata.json [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        "",
+        "Any change in the indexed CmpXchg Values will trigger re-indexing."
+    ],
+
+    "Assets": [
+        
+        {
+            "Title": "Create Compare-Exchange Demo",
+            "Type": "demo",
+            "Url": "/java/compare-exchange/create-compare-exchange"
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/metadata.json:16]: align all `':'`

**Review by Danielle9897 — CHANGES_REQUESTED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/metadata.json (hunk 2 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/metadata.json:156]: this is done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/IndexCompareExchange.java (hunk 3 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/IndexCompareExchange.java:56]: done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/metadata.json (hunk 4 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/compareExchange/indexCompareExchange/metadata.json:16]: done

**Review by Danielle9897 — APPROVED** [review]

**Comment by Danielle9897** [comment]
@ml054 this can be merged

**Pull request merged**